src/database-controller/src/core/k8s.js

```diff
@@ -0,0 +1,250 @@
+    'frameworkcontroller.microsoft.com',
+    'v1',
+    namespace,
+    'frameworks',
+    name,
+  );
+  return res.response;
+}
+
+async function listFramework(name, namespace = 'default') {
+  const res = await customObjectsClient.listNamespacedCustomObject(
+    'frameworkcontroller.microsoft.com',
+    'v1',
+    namespace,
+    'frameworks',
+  );
+  return res.response;
+}
+
+async function createFramework(frameworkDescription, namespace = 'default') {
+  const res = await customObjectsClient.createNamespacedCustomObject(
+    'frameworkcontroller.microsoft.com',
+    'v1',
+    namespace,
+    'frameworks',
+    frameworkDescription,
+  );
+  return res.response;
+}
+
+async function patchFramework(name, data, namespace = 'default') {
+  const res = await customObjectsClient.patchNamespacedCustomObject(
+    'frameworkcontroller.microsoft.com',
+    'v1',
+    namespace,
+    'frameworks',
+    name,
+    data,
+    { headers: { 'Content-Type': 'application/merge-patch+json' } },
+  );
+  return res.response;
+}
+
+async function deleteFramework(name, namespace = 'default') {
+  const res = await customObjectsClient.deleteNamespacedCustomObject(
+    'frameworkcontroller.microsoft.com',
+    'v1',
+    namespace,
+    'frameworks',
+    name,
+  );
+  return res.response;
+}
+
+function getFrameworkInformer(
+  timeoutSeconds = 365 * 86400,
+  namespace = 'default',
+) {
+  /*
+  Usage:
+
+    const informer = getFrameworkInformer()
+    informer.on('add', (obj) => { console.log(`Added: ${obj.metadata.name}`); });
+    informer.on('update', (obj) => { console.log(`Updated: ${obj.metadata.name}`); });
+    informer.on('delete', (obj) => { console.log(`Deleted: ${obj.metadata.name}`); });
+    informer.on('error', (err) => { console.error(err);});
+    informer.start();
+
+  If the informer disconnects normally from API server, it will re-connect automatically.
+  But during the reconnection, listFn will be called again, which is inefficient.
+  According to https://github.com/kubernetes-client/javascript/blob/932c2fbc34db954c6ed397b3cd9ead08b2ff1d10/src/cache.ts#L82-L85,
+  this behavior will be fixed in the future.
+  TO DO: If @kubernetes/client-node fixes this issue, we should upgrade our code to use the new code.
+
+  If the informer encounters any error, it will stop watching, and won't re-connect.
+  One can restart it by informer.on('error', (err) => { informer.start(); }).
+
+  */
+  const listFn = () => {
+    logger.info('Frameworks are listed.');
+    return customObjectsClient.listNamespacedCustomObject(
+      'frameworkcontroller.microsoft.com',
+      'v1',
+      namespace,
+      'frameworks',
+    );
+  };
+  const informer = k8s.makeInformer(
+    kc,
+    `/apis/frameworkcontroller.microsoft.com/v1/frameworks?timeoutSeconds=${timeoutSeconds}`,
+    listFn,
+  );
+  return informer;
+}
+
+const priorityClassClient = kc.makeApiClient(k8s.SchedulingV1Api);
```

Comment:
>priorityClassClient [](start = 6, length = 19)

schedulingV1Client?

Reply:
Refer to https://github.com/kubernetes-client/javascript/blob/b218eb9e7f6080239b7061d42a088749b1d4a749/src/gen/api/schedulingV1Api.ts

Reply:
But there is no k8s.PriorityClassApi?

Reply:
you can search it: https://github.com/kubernetes-client/javascript/search?q=createpriorityClass&unscoped_q=createpriorityClass

Reply:
Can the priorityClassClient create any other objects than PriorityClass in future? If yes, schedulingV1Client is better

Reply:
It only creates priority class, both in the k8s node client and our context.

Reply:
I mean k8s **in future** may create more things by schedulingV1Client

Reply:
I'm not sure whether k8s will use this api group in future. In our context, we only use it to create PriorityClass, so I name it priorityClassClient.

Reply:
ok